contracts/utils/OevSearcherMulticallV1.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./interfaces/IOevSearcherMulticallV1.sol";
+
+/// @title Contract that enables the owner OEV searcher to make batched calls
+/// to external, trusted accounts to facilitate value extraction
+/// @notice Any of the batched calls reverting will result in the transaction
+/// to be reverted. Batched calls are allowed to send values. The contract is
+/// allowed to receive funds in case this is required during value extraction.
+/// @dev OEV searchers that will be targeting the same contracts repeatedly are
+/// recommended to develop and use an optimized version of this contract
+contract OevSearcherMulticallV1 is Ownable, IOevSearcherMulticallV1 {
+    receive() external payable {}
+
+    /// @notice Called by the owner OEV searcher to batch calls with value to
+    /// external, trusted accounts. Any of these calls reverting causes this
+    /// function to revert.
+    /// @dev Calls made to non-contract accounts do not revert. This can be
+    /// used to sweep the funds in the contract.
+    /// @param targets Array of target addresses of batched calls
+    /// @param data Array of calldata of batched calls
+    /// @param values Array of values of batched calls
+    /// @return returndata Array of returndata of batched calls
+    function externalMulticallWithValue(
+        address[] calldata targets,
+        bytes[] calldata data,
+        uint256[] calldata values
+    ) external payable override onlyOwner returns (bytes[] memory returndata) {
+        uint256 callCount = targets.length;
+        require(
+            callCount == data.length && callCount == values.length,
+            "Parameter length mismatch"
+        );
+        returndata = new bytes[](callCount);
+        for (uint256 ind = 0; ind < callCount; ) {
+            bool success;
+            // solhint-disable-next-line avoid-low-level-calls
+            (success, returndata[ind]) = targets[ind].call{value: values[ind]}(
```

Comment:
@bbenligiray Minor but we can cut off 35 gas by caching this

Reply:
```
(success, returndata[ind]) = targets[ind].call{value: values[ind]}(data[ind]);
```
is cheaper than 
```
uint256 value = values[ind];
(success, returndata[ind]) = targets[ind].call{value: value}(data[ind]);
```
The latter is only worthwhile if we use `value` again, and here, it's only used again on the non-happy path because a call failed. So changing this would be optimizing the non-happy path at the expense of the happy path, which is not something we want to do (and the same argument applies to `returndata[ind]`).